sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StandardCoder.java

```diff
@@ -228,4 +232,10 @@ public Object structuralValue(T value) throws Exception {
       }
     }
   }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public TypeDescriptor<T> getEncodedTypeDescriptor() {
+    return (TypeDescriptor<T>) DEFAULT_TYPE_DESCRIPTOR;
```

Comment:
I think you might get an even better default from

```java
TypeDescriptor.of(getClass())
    .resolveType(new TypeDescriptor<T>() {}.getType())
```

As I understand it, if `MyCoder extends StandardCoder<Foo>` then this will actually return a useful `TypeDescriptor<Foo>`.